### PR TITLE
MA-33: Create Admin profile AllUsers/User transaction page

### PR DESCRIPTION
**Pull request by vitalii-petrun**
https://user-images.githubusercontent.com/87064627/219856522-89e2ae67-eaa3-44e3-bade-091521907f0e.mp4

https://user-images.githubusercontent.com/87064627/219858111-f34f56e5-a87a-4ea9-b740-aaff2eec0c46.mp4

### CODE REVIEW

File: lib/presentation/pages/admin_profile/user_transactions_page.dart [hunk 1]
```diff
@@ -1,23 +1,329 @@
...
+import '../../../data/models/transaction.dart';
 import '../../../data/models/user.dart';
 import '../../widgets/profile_header/profile_header.dart';
+import '../../widgets/rights_reserved_footer.dart';
+import '../../widgets/transaction_table/transaction_table.dart';
 
 class UserTransactionsPage extends StatelessWidget {
   final User user;
   final String userName;
+  final bool? showPendingTransactions;
```
Comment (vitalii-petrun) [inline lib/presentation/pages/admin_profile/user_transactions_page.dart:18]: DashboardPage has option to show user's pending transactions. It pushes UserTransactionsPage with already filtered list

File: lib/data/models/user.dart [hunk 2]
```diff
@@ -18,4 +18,13 @@ class User {
...
   });
+
+  User.empty()
+      : name = '',
+        email = '',
+        referralLevel = 0,
+        availableBalance = 0,
+        pendingTransactions = [],
+        transactionsHistory = [],
+        referrals = [];
```
Comment (vovaklh) [inline lib/data/models/user.dart:29]: I think it isn't part of User model

File: lib/l10n/app_en.arb [hunk 3]
```diff
@@ -306,5 +306,19 @@
...
     "@_USER_REFERRALS_PAGE": {},
-    "userReferrals": "User referrals"
+    "userReferrals": "User referrals",
+    "@_TRANSACTIONS_PAGE": {},
+    "allSystemTransactions": "All system transactions",
+    "confirmRequest": "Confirm request",
+    "declineRequest": "Decline request",
+    "userTransactions": "User transactions",
+    "operation": "Operation",
+    "paySystem": "Pay. system",
```
Comment (vovaklh) [inline lib/l10n/app_en.arb:316]: Do we need dot here?

File: lib/presentation/pages/admin_profile/dashboard_page.dart [hunk 4]
```diff
@@ -55,28 +57,46 @@ class _HeaderText extends StatelessWidget {
...
+    final transactions = _generateTransactions();
+
+    return User(
+      name: 'shookones$index',
+      email: 'usermail$index@mail.com',
+      referralLevel: index,
+      availableBalance: Random().nextDouble() * 100000 + 100,
+      pendingTransactions:
+          transactions.where((element) => element.status == 'pending').toList(),
+      transactionsHistory: transactions,
```
Comment (vovaklh) [inline lib/presentation/pages/admin_profile/dashboard_page.dart:70]: I think it should be refactored

File: lib/presentation/pages/admin_profile/transactions_page.dart [hunk 5]
```diff
@@ -8,6 +14,77 @@ class TransactionsPage extends StatelessWidget {
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: ProfileHeader.adminSearch(),
+      body: Portal(
```
Comment (vovaklh) [inline lib/presentation/pages/admin_profile/transactions_page.dart:17]: Wy using empty Portal widget?

File: lib/presentation/pages/admin_profile/transactions_page.dart [hunk 6]
```diff
@@ -8,6 +14,77 @@ class TransactionsPage extends StatelessWidget {
...
+              ),
+            ),
+          ),
+        ),
+      ),
     );
   }
 }
+
+final mockTransactions = [
```
Comment (vovaklh) [inline lib/presentation/pages/admin_profile/transactions_page.dart:52]: Make part of class

File: lib/presentation/pages/admin_profile/user_transactions_page.dart (hunk 1 above)
Comment (vovaklh) [inline lib/presentation/pages/admin_profile/user_transactions_page.dart:18]: Set default to false

File: lib/presentation/pages/admin_profile/user_transactions_page.dart [hunk 7]
```diff
@@ -1,23 +1,329 @@
...
+    this.showPendingTransactions,
     Key? key,
   }) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: ProfileHeader.admin(),
-      body: Text(user.name),
+      body: Portal(
```
Comment (vovaklh) [inline lib/presentation/pages/admin_profile/user_transactions_page.dart:31]: Why usign empty Portal?

File: lib/presentation/pages/admin_profile/user_transactions_page.dart [hunk 8]
```diff
@@ -1,23 +1,329 @@
...
+      __TransactionTableWithFilterState();
+}
+
+class __TransactionTableWithFilterState
+    extends State<_TransactionTableWithFilter> {
+  final _operationFieldController = TextEditingController();
+  final _statusFieldController = TextEditingController();
+  final _paymentSystemFieldController = TextEditingController();
+  final _noteFieldController = TextEditingController();
+  late List<Transaction> transactionsHistory;
```
Comment (vovaklh) [inline lib/presentation/pages/admin_profile/user_transactions_page.dart:93]: Add empty space

File: lib/presentation/pages/admin_profile/user_transactions_page.dart [hunk 9]
```diff
@@ -1,23 +1,329 @@
...
+          transaction.network.toLowerCase().contains(paymentSystem) &&
+          transaction.note.toLowerCase().contains(note);
+    }).toList();
+
+    setState(() => transactionsHistory = result);
+  }
+
+  List<Transaction> _getPendingTransactions() {
+    return widget.transactionsHistory.where((transaction) {
+      return transaction.status.toLowerCase() == 'pending';
```
Comment (vovaklh) [inline lib/presentation/pages/admin_profile/user_transactions_page.dart:113]: Change string to enum

File: lib/presentation/pages/admin_profile/user_transactions_page.dart [hunk 10]
```diff
@@ -1,23 +1,329 @@
...
+  }
+
+  @override
+  void initState() {
+    if (widget.showPendingTransactions) {
+      transactionsHistory = _getPendingTransactions();
+    } else {
+      transactionsHistory = widget.transactionsHistory;
+    }
+    super.initState();
```
Comment (vovaklh) [inline lib/presentation/pages/admin_profile/user_transactions_page.dart:124]: Shoul be called at the first place

File: lib/presentation/widgets/transaction_table/components/transaction_info.dart [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+    required this.leading,
+    Key? key,
+  }) : super(key: key);
+
+  TextStyle transactionTextStyle(BuildContext context) {
+    return context.text.profilePageBody.copyWith(
+      fontSize: 16,
+      fontWeight: FontWeight.bold,
+    );
+  }
```
Comment (vovaklh) [inline lib/presentation/widgets/transaction_table/components/transaction_info.dart:20]: I dont like this. Make private for now

**Review by vovaklh — CHANGES_REQUESTED** [review]

File: lib/l10n/app_en.arb (hunk 3 above)
Comment (vitalii-petrun) [reply lib/l10n/app_en.arb:316]: shortened version of Payment.

Comment (vitalii-petrun) [reply lib/l10n/app_en.arb:316]: image

**Review by vovaklh — APPROVED** [review]

**Pull request merged**